### PR TITLE
OIDC-114 : Allow to set-up multiple client configuration

### PR DESCRIPTION
**Pull request by aubincleme**
This is a second approach to OIDC-114, which is a bit different from what was done on the branch OIDC-114. It uses the same principle, which consists in storing OIDC Client configuration as XWiki objects.

This implementation modifies the component `OIDCClientConfiguration` in order to directly look for OIDC Client configurations in XWiki. These configurations are accessed through their own API in the module `oidc-authenticator-object`.

### CODE REVIEW

**Comment by tmortagne** [comment]
> These configurations are accessed through their own API in the module oidc-authenticator-object

I don't think this is the right location. The point of oidc-authenticator-object was not really to contains all OIDC authentictor related xobjects but to independently expose the definition of an OIDC user so that a completely different OIDC authenticator (so with a completely different way of configuring it) can produce "standard" OIDC users.

**Comment by aubincleme** [comment]
> > These configurations are accessed through their own API in the module oidc-authenticator-object
> 
> I don't think this is the right location. The point of oidc-authenticator-object was not really to contains all OIDC authentictor related xobjects but to independently expose the definition of an OIDC user so that a completely different OIDC authenticator (so with a completely different way of configuring it) can produce "standard" OIDC users.

The goal is quite similar here in this case, but for providing an independent API to manage OIDC client configurations. In that case, a different OIDC authenticator could source its client configs from this API without having to depend on the module `oidc-authenticator`

**Comment by tmortagne** [comment]
> In that case, a different OIDC authenticator could source its client configs from this API without having to depend on the module oidc-authenticator

That would suggest that this other implementation supports everything which is exposed in this configuration, some of which could be very specific to the standard implementation.

**Comment by tmortagne** [comment]
I'm fine having the configuration storage in its own module if you really have a use case, but I feel that at least it should be a different package than the one exposing OIDC user specifications (like oidc-authenticator-user and oidc-authenticator-configuration maybe). If that use case does not exist, yet it can always be done later, WDYT ?

**Comment by polx** [comment]
As a general comment, I have tried to have configuration objects extend configuration objects with the IdentityOAuth => AzureAD => Microsoft365 extension chain and must say that this is rather uneasy and error prone.

I don't see much a win if the configuration is made of just 3-4 strings.
Porting a config because a new software was installed should rather be done by a programme that is dedicated to translation and, hopefully, reviewed and tested by an admin. Back-porting seems not a very common case to me.

**Comment by aubincleme** [comment]
> As a general comment, I have tried to have configuration objects extend configuration objects with the IdentityOAuth => AzureAD => Microsoft365 extension chain and must say that this is rather uneasy and error prone.

I'm unsure to see exactly what you mean : in the current case, there's no OIDC Client configuration object that already exists in XWiki, so it's more about providing one that other implementations could use (and have a somewhat clean API to communicate with the object).

**Comment by polx** [comment]
What I mean, @aubincleme : Let the config objects be application specific and don't try to get a common-ground or shared structure. But maybe try to create translators from existing things.

**Comment by aubincleme** [comment]
I would agree for a couple of configuration options, however here we're talking about 19 properties … Now we can still migrate to the `oidc-authenticator` package, I don't have a strong argument for or against it.

File: oidc-authenticator-configuration/pom.xml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    <xwiki.enforcer.enforce-jakarta.activation.skip>true</xwiki.enforcer.enforce-jakarta.activation.skip>
+    <xwiki.enforcer.enforce-jakarta.mail.skip>true</xwiki.enforcer.enforce-jakarta.mail.skip>
+    <xwiki.enforcer.enforce-jta.skip>true</xwiki.enforcer.enforce-jta.skip>
+  </properties>
+
+  <dependencies>
+    <dependency>
+      <groupId>org.xwiki.platform</groupId>
+      <artifactId>xwiki-platform-oldcore</artifactId>
+      <version>9.11</version>
```
Comment (tmortagne) [inline oidc-authenticator-configuration/pom.xml:49]: Why an older version of oldcore ?

File: oidc-authenticator-configuration/pom.xml [hunk 2]
```diff
@@ -46,7 +46,7 @@
     <dependency>
       <groupId>org.xwiki.platform</groupId>
       <artifactId>xwiki-platform-oldcore</artifactId>
-      <version>9.11</version>
+      <version>${platform.version}</version>
```
Comment (tmortagne) [inline oidc-authenticator-configuration/pom.xml:49]: You should also cleanup all the skips in the `<properties>` which are not needed anymore.

**Comment by tmortagne** [comment]
Looks OK in general but would be nice to have some testing as I'm afraid it can be easy to introduce regressions otherwise.

**Comment by aubincleme** [comment]
Added a couple of unit tests. At this point, I'm confident to merge and I'll probably release tomorrow. @tmortagne if you see something problematic please LMK.

**Comment by tmortagne** [comment]
OK great.

**Pull request merged**